numpy/testing/utils.py

```diff
@@ -674,33 +685,12 @@ def assert_array_compare(comparison, x, y, err_msg='', verbose=True,
     x = array(x, copy=False, subok=True)
     y = array(y, copy=False, subok=True)
 
-    def safe_comparison(*args, **kwargs):
```

Comment:
Well, it would filter a bit more (actuallylikely was first put here by Nathaniel or so for the `==` deprecations), but it seems the only test (after supporting NaT explicitely) that needed extra handling was this one test you comment on before. Not sure, maybe some of the deprecations are finished by now anyway?
Of course this might effect downstream a little here, which is why we did not go this way before after all.